### PR TITLE
Allow devices to determine the viewport for each eye

### PR DESCRIPTION
**Pull request by asajeffrey**
This makes the Window determine the viewports, not the demo appl

### CODE REVIEW

File: gl/src/lib.rs [hunk 1]
```diff
@@ -607,6 +619,7 @@ impl Device for GLDevice {
         unsafe {
             gl::BindFramebuffer(gl::FRAMEBUFFER, framebuffer.gl_framebuffer); ck();
             gl::Viewport(0, 0, framebuffer.texture.size.x(), framebuffer.texture.size.y()); ck();
+            gl::Scissor(0, 0, framebuffer.texture.size.x(), framebuffer.texture.size.y()); ck();
```
Comment (pcwalton) [inline gl/src/lib.rs]: Why do you set the scissor box? (I don't think it does anything unless `GL_SCISSOR_TEST` is enabled anyway, no?)

**Comment by asajeffrey** [comment]
Rebased.

**Comment by asajeffrey** [comment]
Rebased.

File: demo/android/rust/src/lib.rs [hunk 2]
```diff
@@ -167,6 +170,28 @@ impl Window for WindowImpl {
...
+    fn view_box_size(&self, mode: Mode) -> Point2DI32 {
+        let mut width = self.size.x();
+        let height = self.size.y();
+        if let Mode::VR = mode {
+            width = width / 2;
+        }
+        Point2DI32::new(width as i32, height as i32)
+    }
+
+    fn make_current(&mut self, mode: Mode, index: Option<u32>) -> RectI32 {
```
Comment (pcwalton) [inline demo/android/rust/src/lib.rs]: I think this is a bit confusing for programmers who are familiar with OpenGL. `glxMakeCurrent`, `eglMakeCurrent`, etc. just make the context current without setting the viewport. How about just having another method `viewport` that subsumes `view_box_size` and the size calculation of this method? That will also avoid a bit of code duplication (the halving of `width`).

File: demo/android/rust/src/lib.rs [hunk 3]
```diff
@@ -167,6 +170,28 @@ impl Window for WindowImpl {
         Point2DI32::new(0, 0)
     }
 
+    fn view_box_size(&self, mode: Mode) -> Point2DI32 {
```
Comment (pcwalton) [inline demo/android/rust/src/lib.rs]: How about something like:

    fn viewport(&self, view: View) -> Rect2DI32 {
        match view {
            View::Mono => { ... }
            View::Stereo(eye) => { ... }
    }

See below for rationale.

**Comment by asajeffrey** [comment]
Rebased and split out getting the viewport from `make_current`.

**Pull request merged**